webapp/tests/mocha/unit/enketo/medic-xpath-extensions.spec.js

```diff
@@ -380,4 +380,39 @@ describe('medic-xpath-extensions', function () {
       });
     });
   });
+
+  describe('validateIdNumber()', () => {
+    const test = func['cht:luhn'];
+
+    it('should return true for a valid ID number', () => {
+      const idNumber = { t: 'arr', v: [{ textContent: '6505285125086' }] };
+      const result = test(idNumber);
+      assert.strictEqual(result.v, true);
+    });
+    it('should return false for an ID number that does not contain 13 digits', () => {
```

Comment:
As far as I can tell the luhn algorithm doesn't have a length restriction and in other contexts having different lengths would be useful, eg: credit cards have 16. Can we remove this and have length validation done through a separate xpath call?

Reply:
The function also does the job of trimming the string, and ensuring that no other characters other than digits are present. Would it not be an opportune place, after those initial cleanups/checks to also check the intended length?
Rather than having to do all the cleanup again in a separate function. An expected length number could optionally be passed in. Since American Express credit cards have a length of 15 (and interestingly CVV of 4 🤔).

Reply:
I've pushed some updates based on the implementation explained above. If the length validation doesn't meet expectations, I'll be happy to adjust it accordingly.